app/services/trainees/create_from_dttp.rb

```diff
@@ -6,6 +6,9 @@ class CreateFromDttp
     include HasDttpMapping
 
     INVALID_TRN = "999999999"
+
+    TRN_REGEX = /^(\s*[0-9]+[.]?[0-9]*\s*|\s*[.][0-9]+\s*)$/.freeze
```

Comment:
Instead of adding the whitespace to the regex rule, how about matching a stripped version of `dttp_trainee.trn` below on line 143?

By adding it to our regex rule it feels like we're saying whitespace is 'correct'... which strictly it's not.

The outcome might be the same but I feel like keeping the regex strictly correct is nicer. You'll see a lot of codebases with `something.strip` before continuing with other validations.

Reply:
makes sense! I've now modified the regex rule and how I've implemented `strip` , let me know what you think!

Also chatted to Arlston and we agreed the regex can be simpler (we didn't need to account for decimal point for example)